conda_smithy/configure_feedstock.py

```diff
@@ -1245,6 +1245,40 @@ def render_drone(jinja_env, forge_config, forge_dir, return_metadata=False):
     )
 
 
+def azure_build_id_from_token(forge_config):
+    try:
+        # If it fails then we switch to a request using an Azure token.
+        from conda_smithy import azure_ci_utils
+
+        config = azure_ci_utils.AzureConfig(
+            org_or_user=forge_config["azure"]["user_or_org"],
+            project_name=forge_config["azure"]["project_name"],
+        )
+        repo = forge_config["github"]["repo_name"]
+        build_info = azure_ci_utils.get_build_id(repo, config)
+        forge_config["azure"]["build_id"] = build_info["build_id"]
```

Comment:
Please break out large try-blocks into their own functions.  Trys should ideally be one line.

Reply:
Done.